reference_guide/settings_guide.md

```diff
@@ -0,0 +1,154 @@
+
+### EP for Declaring Application Settings
+Settings at the Application level use the `com.intellij.applicationConfigurable` EP.
+
+An example `<applicationConfigurable>` EP declaration is shown below.
+See [Settings EP Attributes](#settings-ep-attributes) for details.
+```xml
+  <extensions defaultExtensionNs="com.intellij">
+    <applicationConfigurable parentId="tools" instance="org.company.ApplicationSettingsConfigurable"
+            id="org.company.ApplicationSettingsConfigurable" displayName="My Application Settings"/>
+  </extensions>
+``` 
+
+### EP for Declaring Project Settings
+Project level Settings use the `com.intellij.projectConfigurable` EP.
+
+An example `<projectConfigurable>` EP declaration is shown below.
+See [Settings EP Attributes](#settings-ep-attributes) for details.
+```xml
+  <extensions defaultExtensionNs="com.intellij">
+    <projectConfigurable parentId="tools" instance="org.company.ProjectSettingsConfigurable"
+            id="org.company.ProjectSettingsConfigurable" displayName="My Project Settings"
+            nonDefaultProject="true"/>
+  </extensions>
+``` 
+
+### Settings EP Attributes 
+Readers are encouraged to review the Javadoc comments for [`Configurable`](upsource:///platform/platform-api/src/com/intellij/openapi/options/Configurable.java).
+These comments apply to Settings EP declarations in general, not just for implementations based on the `Configurable` interface.
+This section provides some additional clarification of those comments.
+
+#### Table of Attributes
+The attributes supported by `com.intellij.applicationConfigurable` and `com.intellij.projectConfigurable` EPs are in the table below:
+
+| Attribute | Implementation<br>Basis |Required |&ensp; &ensp; Attribute Value |
+| :---------- | :-----: | :--------: |:------ |
+| `instance` | `Configurable` | Y | FQN of implementation. See [The Configurable Interface](#the-configurable-interface) for more information. |
+| `provider` | `ConfigurableProvider` | Y | FQN of implementation. See [The ConfigurableProvider Class](#the-configurableprovider-class) for more information. |
+| `nonDefaultProject` | `Configurable` | Y | Applicable _only_ to the `com.intellij.projectConfigurable` (project Settings) EP.<br>`true` = show Settings for all projects _except_ the [default project](https://www.jetbrains.com/help/idea/configure-project-settings.html#new-default-settings).<br>`false` = show Settings for all projects. |
+| `displayName` | `Configurable`<br>`ConfigurableProvider` | (1) | The non-localized Settings name visible to users, which is needed for the Settings dialog left-side menu.<br>Providing a Settings display name is highly recommended. If it isn’t declared, the component gets instantiated to retrieve its name dynamically. Loading the plugin's classes increases the delay before showing the Settings dialog.<br>For a _localized_ visible name omit `displayName` and use the `key` and `bundle` attributes. |
```

Comment:
IMHO we can skip the explanation about performance here, it is given above already in general